### ISSUE TITLE
Crash report

### ISSUE THREAD

**Issue by peternlewis** [ClaudeMetz/FactoryPlanner#342]
### Problem Description

Opened the Factory Planner, added a recipe, added a title, added a blue science demand, clicked the demand, and the server crashed.

```
603811.869 Error MainLoop.cpp:1404: Exception at tick 1770376: The mod Factory Planner (1.1.79) caused a non-recoverable error.
Please report this error to the mod author.

Error while running event factoryplanner::Custom event (ID 211)
LuaGuiElement API call when LuaGuiElement was invalid.
stack traceback:
	[C]: in function '__index'
	__factoryplanner__/ui/base/modal_dialog.lua:173: in function 'enter'
	__factoryplanner__/ui/event_handler.lua:209: in function <__factoryplanner__/ui/event_handler.lua:189>
	[C]: in function 'raise_event'
	__factoryplanner__/util/raise.lua:20: in function 'open_dialog'
	__factoryplanner__/ui/main/title_bar.lua:113: in function 'handler'
	__factoryplanner__/ui/event_handler.lua:157: in function <__factoryplanner__/ui/event_handler.lua:114>
stack traceback:
	[C]: in function 'raise_event'
	__factoryplanner__/util/raise.lua:20: in function 'open_dialog'
	__factoryplanner__/ui/main/title_bar.lua:113: in function 'handler'
	__factoryplanner__/ui/event_handler.lua:157: in function <__factoryplanner__/ui/event_handler.lua:114>
603811.869 Error ServerMultiplayerManager.cpp:92: MultiplayerManager failed: "The mod Factory Planner (1.1.79) caused a non-recoverable error.
Please report this error to the mod author.

Error while running event factoryplanner::Custom event (ID 211)
LuaGuiElement API call when LuaGuiElement was invalid.
stack traceback:
	[C]: in function '__index'
	__factoryplanner__/ui/base/modal_dialog.lua:173: in function 'enter'
	__factoryplanner__/ui/event_handler.lua:209: in function <__factoryplanner__/ui/event_handler.lua:189>
	[C]: in function 'raise_event'
	__factoryplanner__/util/raise.lua:20: in function 'open_dialog'
	__factoryplanner__/ui/main/title_bar.lua:113: in function 'handler'
	__factoryplanner__/ui/event_handler.lua:157: in function <__factoryplanner__/ui/event_handler.lua:114>
stack traceback:
	[C]: in function 'raise_event'
	__factoryplanner__/util/raise.lua:20: in function 'open_dialog'
	__factoryplanner__/ui/main/title_bar.lua:113: in function 'handler'
	__factoryplanner__/ui/event_handler.lua:157: in function <__factoryplanner__/ui/event_handler.lua:114>"
603811.869 Info ServerMultiplayerManager.cpp:816: updateTick(1770376) changing state from(InGame) to(Failed)
603811.869 Quitting: multiplayer error.
603812.138 Info ServerMultiplayerManager.cpp:154: Quitting multiplayer connection.
603812.138 Info ServerMultiplayerManager.cpp:816: updateTick(4294967295) changing state from(Failed) to(Closed)
603812.139 Info HttpSharedState.cpp:58: Downloading https://multiplayer.factorio.com/remove-game/18617527
603813.248 Info UDPSocket.cpp:233: Closing socket
603813.248 Info UDPSocket.cpp:263: Socket closed
603813.248 Info UDPSocket.cpp:233: Closing socket
603813.320 Info UDPSocket.cpp:233: Closing socket
603813.320 Info UDPSocket.cpp:263: Socket closed
603813.321 Goodbye
proxy:~/factorio% 
```

### Reproduction

Not sure yet.

**Comment by peternlewis** [comment]
Restarted the server, tried again, server did not crash. So not immediately repeatable.

**Comment by ClaudeMetz** [comment]
Hi, yeah this is some kind of issue related to multiplayer latency, which is very annoying. It's a duplicate of #219, which I'll try to fix in some way for 2.0. Thanks for the report.
